resources.tf

```diff
@@ -217,3 +223,48 @@ resource "aws_iam_role_policy_attachment" "emr_analytical_dataset_write_s3" {
   role       = aws_iam_role.analytical_dataset_generator.name
   policy_arn = aws_iam_policy.analytical_dataset_write_s3.arn
 }
+
+
+# Security groups
+
+resource "aws_security_group" "analytical_dataset_generation" {
+  name                   = "analytical_dataset_generation_common"
+  description            = "Contains rules for both EMR cluster master nodes and EMR cluster slave nodes"
+  revoke_rules_on_delete = true
+  #vpc_id                 = module.vpc.vpc_id
+}
+
+resource "aws_security_group_rule" "analytical_dataset_generation_egress" {
+  description       = "Allow outbound traffic from Analytical Dataset Generation EMR Cluster"
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  #prefix_list_ids   = [module.vpc.s3_prefix_list_id]
+  security_group_id = aws_security_group.analytical_dataset_generation.id
+}
+
+resource "aws_security_group_rule" "analytical_dataset_generation_ingress" {
+  description       = "Allow inbound traffic from Analytical Dataset Generation EMR Cluster"
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  #prefix_list_ids   = [module.vpc.s3_prefix_list_id]
```

Comment:
This will need exporting from the aws-ingestion state file, so agreed that we should hold off on this for now.
You will need a `cidr_blocks` attribute instead though as an sg rule requires knowing something about the source/destination network. Same for the sg rule above.